hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -246,15 +246,8 @@ public OmSnapshotManager(OzoneManager ozoneManager) {
       }
     };
 
-    // init LRU cache
-    this.snapshotCache = CacheBuilder.newBuilder()
-        // Indicating OmSnapshot instances are softly referenced from the cache.
-        // If no thread is holding a strong reference to an OmSnapshot instance
-        // (e.g. SnapDiff), the instance could be garbage collected by JVM at
-        // its discretion.
-        .softValues()
```

Comment:
Hi @GeorgeJahad , the plan was to do `softValues()` first (because that was much easier and quicker to impl), then do custom cache impl to entirely replace the `LoadingCache` (as the title suggests).

We recently are suspecting a small issue with `softValues()`, where seemingly too many instances are open at a time and each are holding quite a few files open. Thus we filed #4884 as a mitigation.

It is plausible to make the interface pluggable. But with the only advantage of rolling back to `LoadingCache` impl when custom cache impl goes wrong. Would you like to file a new jira for this? (requires adding a new OM config, adding a new interface, tweaking `RC#close` / `decrementRefCount`, etc.)

Reply:
> Would you like to file a new jira for this? 
Nope, just wanted to clarify.  (I didn't remember us saying we would replace the other approach.)

But now I'm confused about why we are using com.google.common.cache.CacheLoader and  com.google.common.cache.RemovalListener.  Do we still need those?

Reply:
> But now I'm confused about why we are using com.google.common.cache.CacheLoader and com.google.common.cache.RemovalListener. Do we still need those?

We don't exactly need those anymore. I kept them from the beginning just because it would be less disruptive (less conflicts) if others are also fixing something in the loader at the same time. But this also leaves us the option to implement a pluggable interface if we need to.

Reply:
ok, but now I'm not quite sure how the underlying cache is stored.  Does it have it's own eviction policy.

Also, I find it confusing to have the removalListener implemented but not having it be used anywhere.

Reply:
Ok I'm going to add a TODO on `removalListener`. So that in the follow-up PR we can decide whether to keep it (make cache interface pluggable), or remove it (make custom SnapshotCache the only solution).

Reply:
With this PR, actual cache structure is `dbMap` in custom `SnapshotCache`. It is just utilizing the existing `CacheLoader` as the loader. Can either keep or remove it when we decide whether we want to make the cache impl pluggable.